components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/service/impl/WebhookManagementServiceImpl.java

```diff
@@ -167,8 +139,19 @@ public void deleteWebhook(String webhookId, String tenantDomain) throws WebhookM
                     webhookId, tenantDomain));
         }
         int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
-        if (isWebhookExists(webhookId, tenantId)) {
-            daoFACADE.deleteWebhook(webhookId, tenantId);
+        Webhook existingWebhook = daoFACADE.getWebhook(webhookId, tenantId);
+        if (existingWebhook == null) {
+            throw WebhookManagementExceptionHandler.handleClientException(
+                    ErrorMessage.ERROR_CODE_WEBHOOK_NOT_FOUND, webhookId);
+        } else {
+            if ((existingWebhook.getStatus() == WebhookStatus.INACTIVE) ||
+                    (existingWebhook.getStatus() == WebhookStatus.PENDING_DEACTIVATION)) {
+                daoFACADE.deleteWebhook(webhookId, tenantId);
+            } else if ((existingWebhook.getStatus() == WebhookStatus.ACTIVE) ||
+                    (existingWebhook.getStatus() == WebhookStatus.PENDING_ACTIVATION)) {
+                throw WebhookManagementExceptionHandler.handleClientException(
+                        ErrorMessage.ERROR_CODE_WEBHOOK_DELETE_NOT_ALLOWED_ERROR, webhookId);
+            }
```

Comment:
Rather than branching have if checks with which you can check and throw first and move the real operation to the latter part

Reply:
Refactored and improved the readability
fixed with https://github.com/wso2/carbon-identity-framework/pull/6885/commits/fb575c056228c93d2221b4ea29c500a76d09311c